packages/api-client/src/importData.js

```diff
@@ -17,42 +19,107 @@ const getRemoteEntityId = (localData: Object, fieldSchema: Object, $id: string,
   return id;
 };
 
+const MAX_PARALLEL_CREATIONS = 50;
+
+const uploadFiles = async (record, tableSchema, filestackClient, path) => {
+  const fieldNames = R.keys(record);
+
+  let nextRecord = record;
+
+  for (let i = fieldNames.length - 1; i >= 0; i--) {
+    const fieldName = fieldNames[i];
+
+    const fieldSchema = getFieldSchemaByName(fieldName, tableSchema);
+
+    if (fieldSchema && isFileField(fieldSchema)) {
+      if (isListField(fieldSchema)) {
+        if (Array.isArray(record[fieldName])) {
+          for (let j = 0; j < record[fieldName].length; j++) {
+            nextRecord = R.assocPath([fieldName, j, 'fileId'], (await (filestackClient.storeURL(record[fieldName][j].url, {
+              path,
+            }))).handle, nextRecord);
+            nextRecord = R.dissocPath([fieldName, j, 'url'], nextRecord);
+          }
+        }
+      } else {
+        if (record[fieldName]) {
+          nextRecord = R.assocPath([fieldName, 'fileId'], (await (filestackClient.storeURL(record[fieldName].url, {
+            path,
+          }))).handle, nextRecord);
+          nextRecord = R.dissocPath([fieldName, 'url'], nextRecord);
+        }
+      }
+    }
+  }
+
+  return nextRecord;
+};
+
 export const importData = async (request: (query: string | DocumentNode, variables?: Object) => Promise<Object>, schemaData: Object) => {
   const { tablesList: { items: tableSchema }} = await request(TABLES_LIST_QUERY, {
     filter: {
       onlyUserTables: false,
     },
   });
 
+  let fileUploadInfo = {};
+
+  try {
+    ({ fileUploadInfo } = await request(FILE_UPLOAD_INFO_QUERY));
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.log('Can\'t fetch file upload info', e);
```

Comment:
Maybe warning? Maybe add abstraction for logs?

Reply:
@ej9x it's the experimental functional in the future we should throw the specific error for this case